tools/clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -466,8 +466,11 @@ llvm::DIType *CGDebugInfo::CreateType(const BuiltinType *BT) {
   case BuiltinType::OCLEvent:
     llvm_unreachable("No ObjC or OpenCL support");
   case BuiltinType::LitInt:
+    Encoding = llvm::dwarf::DW_ATE_signed;
+    break;
   case BuiltinType::LitFloat:
-    llvm_unreachable("Unsupported HLSL types");
+    Encoding = llvm::dwarf::DW_ATE_float;
+    break;
```

Comment:
The fault really lies with me for adding these here in the first place, but a better place for these would be the HLSL-specific sections of cases below including Min16Int  and HalfFloat respectively. This `#else` block is meant for ObjC and OCL stuff.

Reply:
fixed